### PR TITLE
Messaging UI and IN

### LINKED ISSUE #34: IN/FE: Messaging
- Create a chat UI.
- A left sidebar that shows conversations with users, a chat UI to display messages and allow sending messages on the right
- Integrate with backend routes to send/receive messages

### PR DESCRIPTION
**Pull request by g4rry420**
### What this PR does (required):
- Added responsive messaging UI
- Integrated it with backend also.
- Also, this pr depends on #57. First, we should merge that one.

### Screenshots / Videos (required):
messagUI1
messagUI3

### Any information needed to test this feature (required):
- First a create conversation using Postman in the BE at `/convo/` post route with array of userIds.
- At FR click on the `Messages` navbar tab.

### Any issues with the current functionality (optional):
No

### CODE REVIEW

**Comment by dproc96** [comment]
It looks like there are a lot of changes from #57 in this which makes the diff really hard to parse, can you change the base branch of this to 57?

**Comment by g4rry420** [comment]
> It looks like there are a lot of changes from #57 in this which makes the diff really hard to parse, can you change the base branch of this to 57?

@dproc96  Done

File: client/src/pages/Messages/ActiveChat/ActiveChat.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import { useAuth } from '../../../context/useAuthContext';
+import { useMessage } from '../../../context/useMessageContext';
+
+export interface IMessages {
+  _id: string;
+  author: string;
+  text: string;
+  createdAt: string;
+  updatedAt: string;
+}
```
Comment (moffatethan) [inline client/src/pages/Messages/ActiveChat/ActiveChat.tsx]: Couldn't you move this to interfaces folder and use this both in the context and tsx files?

Comment (moffatethan) [reply client/src/pages/Messages/ActiveChat/ActiveChat.tsx]: I don't see this even being used in this file, remove?

File: client/src/pages/Messages/ActiveChat/Input.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  const [text, setText] = useState<string>('');
+  const classes = useStyles();
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => setText(event.target.value);
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (text === '') return;
+    // add sender user info if posting to a brand new convo, so that the other user will have access to username, profile pic, etc.
+    const message = {
+      _id: Math.random().toString(),
```
Comment (moffatethan) [inline client/src/pages/Messages/ActiveChat/Input.tsx]: This is not a good way to set IDs, if you aren't relying on MongoIDs then I recommend using `nanoid` or `uuid`

File: client/src/pages/Messages/ActiveChat/Input.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      text: text,
+      author: currentUserId,
+      createdAt: new Date().toISOString(),
+      updatedAt: new Date().toISOString(),
```
Comment (moffatethan) [inline client/src/pages/Messages/ActiveChat/Input.tsx]: In your schema file, you have set timestamps. This would not be needed to pass down as Mongo will automatically generate these fields for us

**Review by moffatethan — COMMENTED** [review]
Left you some comments to address. It's looking good

File: client/src/pages/Messages/ActiveChat/ActiveChat.tsx (hunk 1 above)
Comment (g4rry420) [reply client/src/pages/Messages/ActiveChat/ActiveChat.tsx]: Sure, I do that

File: client/src/pages/Messages/ActiveChat/Input.tsx (hunk 2 above)
Comment (g4rry420) [reply client/src/pages/Messages/ActiveChat/Input.tsx]: I will change it.

File: client/src/pages/Messages/ActiveChat/Input.tsx (hunk 3 above)
Comment (g4rry420) [reply client/src/pages/Messages/ActiveChat/Input.tsx]: This was actually used internally between the sockets while passing data `createdAt` field is required but I will remove `updatedAt`.

Comment (moffatethan) [reply client/src/pages/Messages/ActiveChat/Input.tsx]: Okay! 👍🏻

**Review by moffatethan — APPROVED** [review]
LGTM!
